django/urls/conf.py

```diff
@@ -70,6 +72,14 @@ def _path(route, view, kwargs=None, name=None, Pattern=None):
         pattern = Pattern(route, name=name, is_endpoint=True)
         return URLPattern(pattern, view, kwargs, name)
     else:
+        # Detect the user instantiating the CBV instead of calling .as_view()
+        if isinstance(view, View):
+            view_cls = view.__class__
+            raise ImproperlyConfigured(
+                "URL route '%s' should pass in '%s.as_view()' instead of '%s()'" %
+                (route, view_cls.__name__, view_cls.__name__)
+            )
+
```

Comment:
This too. I meant for both of the checks to be via `check()`.

Reply:
It's not possible to move this, an exception already gets raised immediately after this if statement, making it impossible to finish constructing the URLPattern instance that we could later invoke check() on.

Reply:
I see. I think then branch on the `message` and keep the raised error the same. (It makes no sense that we have `ImproperlyConfigured` for one case and `TypeError` for another.)